anndata/core/raw.py

```diff
@@ -16,20 +16,28 @@
 class Raw:
     def __init__(
         self,
-        adata: Optional['anndata.AnnData'] = None,
+        adata: 'anndata.AnnData',
         X: Union[np.ndarray, sparse.spmatrix, None] = None,
-        var: Union[AxisArrays, AxisArraysView, None] = None,
-        varm: Union[AxisArrays, AxisArraysView, None] = None,
+        var: Union[
+            AxisArrays, AxisArraysView, Mapping[str, Sequence], None
+        ] = None,
+        varm: Union[
+            AxisArrays, AxisArraysView, Mapping[str, np.ndarray], None
+        ] = None,
     ):
+        from .anndata import _gen_dataframe
+
         self._adata = adata
         self._n_obs = adata.n_obs
         if X is not None:
             self._X = X
-            self._var = var
+            self._var = _gen_dataframe(var, X.shape[1], ['var_names'])
             self._varm = AxisArrays(self, 1, varm)
         else:
             self._X = None if adata.isbacked else adata.X.copy()
-            self._var = adata.var.copy()
+            self._var = _gen_dataframe(
+                adata.var.copy(), adata.n_vars, ['var_names']
+            )
```

Comment:
If this is going to come from `adata.var.copy()` (which is a weird thing to happen, and seems wrong), then it would have already been through validation and wouldn't need `_gen_dataframe`,

Reply:
Right!

Reply:
Isn't this the case where it's backed? Why would it copy the parents `.var` when it's backed, but not otherwise?

Reply:
Haha, I'm probably not the right person to ask about expectations for `.raw`. I do not like that it exists, and think it should go since we can just do `layers` or a separate object.

I think it's behaviour should line up with AnnData as much as possible. Since backed mode didn't really work before (in many cases). I'm hoping we're okay to change this.

I'm wouldn't think that it's possible for `.X` to be specified without `.var` since there should at least be a `var_names`.